File: app/src/main/java/com/rakib/foodbankbd/MainActivity.java
package com.rakib.foodbankbd;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.pm.ActivityInfo;
import android.os.Bundle;

import com.rakib.foodbankbd.adapters.AreaRecyclerViewAdapter;
import com.rakib.foodbankbd.entities.AreaObject;
import com.rakib.foodbankbd.helpers.SpacesItemDecoration;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private RecyclerView areaRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        areaRecyclerView = findViewById(R.id.recyclerViewArea);

        GridLayoutManager gridLayoutManager = new GridLayoutManager(MainActivity.this, 2);
        areaRecyclerView.setLayoutManager(gridLayoutManager);
        areaRecyclerView.setHasFixedSize(true);
        areaRecyclerView.addItemDecoration(new SpacesItemDecoration(2, 24, true));
        AreaRecyclerViewAdapter areaRecyclerViewAdapter = new AreaRecyclerViewAdapter(MainActivity.this, getAllArea());
        areaRecyclerView.setAdapter(areaRecyclerViewAdapter);
    }

    private List<AreaObject> getAllArea() {
        List<AreaObject> areas = new ArrayList<AreaObject>();
        areas.add(new AreaObject("Shyamoli", R.drawable.shyamoli));
        areas.add(new AreaObject("Dhanmondi", R.drawable.dhanmondi));
        areas.add(new AreaObject("Banani", R.drawable.banani));
        areas.add(new AreaObject("Mirpur", R.drawable.mirpur));
        areas.add(new AreaObject("Wari", R.drawable.wari));
        return areas;
    }
}

File: app/src/main/java/com/rakib/foodbankbd/MainActivity.kt
package com.rakib.foodbankbd

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import com.rakib.foodbankbd.adapters.AreaRecyclerViewAdapter
import com.rakib.foodbankbd.entities.AreaObject
import com.rakib.foodbankbd.helpers.SpacesItemDecoration
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val gridLayoutManager = GridLayoutManager(this@MainActivity, 2)
        recyclerViewArea.layoutManager = gridLayoutManager
        recyclerViewArea.setHasFixedSize(true)
        recyclerViewArea.addItemDecoration(SpacesItemDecoration(2, 24, true))
        val areaRecyclerViewAdapter = AreaRecyclerViewAdapter(this@MainActivity, allArea)
        recyclerViewArea.adapter = areaRecyclerViewAdapter
    }

    private val allArea: List<AreaObject>
        get() {
            val areas: MutableList<AreaObject> = ArrayList()
            areas.add(AreaObject("Shyamoli", R.drawable.shyamoli))
            areas.add(AreaObject("Dhanmondi", R.drawable.dhanmondi))
            areas.add(AreaObject("Banani", R.drawable.banani))
            areas.add(AreaObject("Mirpur", R.drawable.mirpur))
            areas.add(AreaObject("Wari", R.drawable.wari))
            return areas
        }
}